newIDE/app/src/EffectsList/EffectDescription.js

```diff
@@ -11,8 +11,6 @@ export type EffectDescription = {|
   parameterDefaultValues: Array<EffectDefaultValue>,
 |};
 
-const clampValue = value => Math.max(0, Math.min(1, value));
```

Comment:
Actually, both the editor and the game engine carry the responsibility of validating the user input, but for two different reasons:
- The game engine because we don't want to pass invalid values to the filters - that could make strange results or even things like crashs (in other words, no bugs).
- The editor because it's a bad user experience for the user to be able to enter invalid values - if we know about the valid values we should as possible avoid to store them.

It's of course entirely possible for the user to enter invalid values using events, but that's why we have the checks in the game engine. The second checks are the editor best effort to avoid you entering non sense things :)

Reply:
So should we keep a second copy of the validation functions for the editor?

Reply:
Yes let's keep a copy of it, one in the game engine and one in the editor. Will generally in programming, it's best to factor things, here it's better to have both of them because there serves a slightly different purpose and it's not harmful. Better have a copy rather than a bad abstraction :)

> Is it possible to check for valid values while entering them in the textbox?

Not really, the text fields are made so that changes are applied only when they are losing the focus. Doing validation during the writing is trickier and can lead to strange results (like, you enter an invalid value like 4, and it's getting correct immediately => people can think of a bug).